server/src/main/java/com/skillmagnet/Course/CourseController.java

```diff
@@ -0,0 +1,126 @@
+package com.skillmagnet.Course;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+@RequestMapping("api")
+public class CourseController {
+    
+    /**
+     * hooks up to sql db for operations
+     */
+    @Autowired
+    CourseRepository courseRepository;
+
+    /**
+     * Retrieves specific course by id
+     * @param id - id of course to find
+     * @return - course object || NOT_FOUND
+     */
+    @GetMapping("/course")
```

Comment:
Hate to sound like a nitpicker here lol but definitely I think is beneficial to have in mind

But generally with REST API styles, you'd want to use a path variable to identity resources and request/query params to do stuff like filtering, paging, etc. Here's an [example](https://www.baeldung.com/spring-pathvariable) of what I mean.

So to get a specific course, it's like hey I want courses, specifically with ID 7 so `GET /course/7`
But if it's hey I just want courses, then `GET /course`

But then the extras (so paging, filtering) are tacked onto the request. So get me all courses (the resource) where the name contains 'Java'. `GET /course?filter=java`

Reply:
Pretty sure I understand what you're saying. I went through and updated the URL's and am using `@PathVariable` now. Definitely makes sense and is more simplified.